charger/nrggen2.go

```diff
@@ -0,0 +1,365 @@
+package charger
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/modbus"
+	"github.com/evcc-io/evcc/util/sponsor"
+	"github.com/spf13/cast"
+	"github.com/volkszaehler/mbmd/encoding"
+)
+
+// https://www.nrgkick.com/wp-content/uploads/2024/07/local_api_docu_simulate.html
+
+// NRGKickGen2 charger implementation
+type NRGKickGen2 struct {
+	conn *modbus.Connection
+}
+
+const (
+	// All register use LittleEndian
+	// Read only (0x03)
+	nrgKickGen2Serial            = 0  // 11 regs
+	nrgKickGen2ModelType         = 11 // 16 regs
+	nrgKickGen2MaxPhases         = 36
+	nrgKickGen2SoftwareVersionSM = 122 // 8 regs
+	// Read (0x03) / Write (0x06, 0x16) Registers
+	nrgKickGen2ChargingCurrent = 194 // A, factor 10
+	nrgKickGen2Enabled         = 195
+	nrgKickGen2Phases          = 198
+	// Read only (0x03)
+	nrgKickGen2TotalChargedEnergy = 199 // Wh, 4 regs
+	nrgKickGen2ChargedEnergy      = 203 // Wh, 2 regs
+	nrgKickGen2TotalActivePower   = 210 // W, 2 regs, factor 1000
+	nrgKickGen2PhaseVoltages      = 217 // factor 100
+	nrgKickGen2PhaseCurrents      = 220 // factor 1000
+	nrgKickGen2RegStatus          = 251
+	nrgKickGen2RegRelais          = 253
+	nrgKickGen2RegRCD             = 255
+	nrgKickGen2RegWarning         = 256
+	nrgKickGen2RegError           = 257
+)
+
+func init() {
+	registry.Add("nrgkick-gen2", NewNRGKickGen2FromConfig)
+}
+
+//go:generate go run ../cmd/tools/decorate.go -f decorateNRGKickGen2 -b *NRGKickGen2 -r api.Charger -t "api.ChargerEx,MaxCurrentMillis,func(float64) (error)" -t "api.PhaseSwitcher,Phases1p3p,func(int) error"
+
+// NewNRGKickGen2FromConfig creates a NRGKickGen2 charger from generic config
+func NewNRGKickGen2FromConfig(other map[string]interface{}) (api.Charger, error) {
+	cc := modbus.TcpSettings{
+		ID: 1,
+	}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	nrg, err := NewNRGKickGen2(cc.URI, cc.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	// detect configurable features
+	var (
+		chargerEx func(float64) error
+		phasesS   func(int) error
+	)
+
+	enabled, err := nrg.Enabled()
+	if err != nil {
+		return nil, err
+	}
+
+	err = nrg.Enable(false)
```

Comment:
That's nothing we should do when starting!

Reply:
the 1p3p and mA capabilities are now configurable via yaml template

Reply:
Das wäre die letzte Option. Lassen sich die Optionen wirklich nicht auslesen?

Reply:
Ich hab in den offiziellen API Beschreibungen leider keine Hinweise darüber gefunden. Siehe Link im Pull Request

Reply:
Könnte man den Strom testen? Also auslesen, und falls nciht auf .1A korrekt mal um +.1A schreibenm dann schauen ob das wieder eingelesen werden kann? 

Phasen konfigurierbar ist nachvollziehbar, Strom schon arg unglücklich. 

Gibts denn einen Grund anzunehmen, dass nciht _alle_ gen2 das können? Evtl. mal den Support fragen? Leider scheint es, als würde die Arbeit mit diesen Boxen sehr mühsam bleiben :(

Reply:
oh geez... Ich habs eben getestet. Man kann .1A setzen auch wenn die Funktion via App deaktiviert ist. Es scheint also als würde es immer einstellbar sein, nur via App ist die Funktion dann deaktiviert. Phasenumschaltung ist aber grundsätzlich deaktiviert. Den Hinweis in der Schnittstellendokumentation findet man ebenso dass man es zuvor via App aktivieren muss.
Ich schmeiss .1A Cap wieder raus und aktivier es dauerhaft.

Reply:
hast du noch Empfehlungen für Tests?
ich nutze aktuell lediglich den `evcc charger` CLI Befehl. Damit kann ich leider keine .1A setzen. Via UI bin ich derzeit überfordert wie ich einen Test durchführen könnte

Reply:
Probier gerne mal https://github.com/evcc-io/evcc/pull/15341

Reply:
funktioniert, danke :)